backup/main.py

```diff
@@ -2,51 +2,92 @@
-from sql_queries import (
-    FETCH_MEAL_RECORDS,
-    DELETE_MEAL_RECORDS,
-    FETCH_PANTRY_RECORDS,
-    DELETE_PANTRY_RECORDS,
-)
-from slack_service import send_message
-from csv_service import create_csv
-from db import execute_query
-from gdrive import upload_to_drive
-
-
-def handle_ops(fetch_query, delete_query):
-    records = execute_query(fetch_query)
-    print(fetch_query)
+import psycopg2
+from dotenv import find_dotenv, load_dotenv
 
+import csv_service
+import db
+import gdrive
+from slack_service import SlackService, make_slack_client
+from sql_queries import (DELETE_MEAL_RECORDS, DELETE_PANTRY_RECORDS,
+                         FETCH_MEAL_RECORDS, FETCH_PANTRY_RECORDS)
+
+load_dotenv(find_dotenv())
+
+
+def handle_ops(db_connection, fetch_query, delete_query, **kwargs):
+    print("Executing fetch_query...")
+    records = db.execute_query(db_connection, fetch_query, fetch=True)
+    slack_service = kwargs.get("slack_service")
     if records:
-        print("There are records.")
-        csv_path, filename = create_csv(records)
-        execute_query(delete_query)
-        upload_to_drive(csv_path, filename)
+        print(f"{len(records)} records found.")
+        print("writing records to csv file...")
+        csv_path, filename = csv_service.create_csv(records)
+        # enable/disable the delete action
+        if kwargs.get("enable_delete", False) is True:
+            print("Executing delete_query...")
+            db.execute_query(db_connection, delete_query)
+        else:
+            print("skipping delete, it is a dangerous action :)")
+        # enable/disable gdrive upload
+        if kwargs.get("enable_gdrive_upload") is True:
+            print("Uploading csv file to GoogleDrive...")
+            gdrive.upload_to_drive(csv_path, filename)
+        else:
+            print("skipping gdrive upload, no GDRIVE_TOKEN set")
+        print("Cleaning up csv files...")
         os.remove(csv_path)
     else:
         print("There are no records for this query.")
-        send_message("There are no records for this query.")
+        slack_service.send_message("There are no records for this query.")
+
+
+def local_chat_post_message(**kwargs):
+    """Dummy function that does nothing when SLACK_TOKEN is not set."""
+    return {}
 
 
 def run():
     try:
         today = date.today()
+        enable_delete = os.getenv("ENABLE_BACKUP_DELETE_ACTION")
+        enable_gdrive_upload = os.getenv("GDRIVE_TOKEN", "") != ""
 
-        if today.day > 1:
-            # fail silently because it's not the first of the month
-            send_message("It's not yet time for backup")
+        channel = os.getenv("SLACK_CHANNEL")
+        client = make_slack_client(local_chat_post_message)
+        slack_service = SlackService(client, channel)
+
+        if today.day > int(os.getenv("BACKUP_DAY")):
+            # fail silently because it's not the 1st of the month
+            slack_service.send_message("It's not yet time for backup")
```

Comment:
We'll remove this method because we'll potentially have to run this method every day since not all months have the same number of days.

So we'll just return silently so nothing happens. Make sense?